stdlib/public/SwiftShims/MetadataSections.h

```diff
@@ -59,17 +64,12 @@ struct MetadataSections {
   /// \sa swift_addNewDSOImage()
   const void *baseAddress;
 
-  /// `next` and `prev` are used by the runtime to construct a
```

Comment:
This comment is very concerned with heap allocations but `swift_addNewDSOImage()` immediately calls a number of functions (e.g. `addImageProtocolsBlockCallback()`) that can allocate by way of their own `ConcurrentReadableArray` structures. 🤷

Reply:
The last portion of it is the concern: this occurs in a constructor, which means that malloc may not be yet setup.  We should have two phased initialization: one to register all the modules, and one to actually update the runtime state after all the modules are loaded.

Reply:
Could this be resolved by forcing the Swift-generated constructor functions to have lower priority (run after) the default? Presumably libmalloc or equivalent is going to be set up by the time default-priority constructors have all finished.

Reply:
(The equivalent for Windows being [`init_seg(lib)`](https://docs.microsoft.com/en-us/cpp/preprocessor/init-seg?view=msvc-170) I think?)

Reply:
Given that this is already the behaviour of the Swift runtime, this isn't a regression. We should track a fix in a separate PR.